packages/react-theme-provider/src/ThemeProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { useStylesheet } from '@fluentui/react-stylesheets';
+import { tokensToStyleObject } from './tokensToStyleObject';
+import { ThemeContext } from './ThemeContext';
+import { Theme, ThemePrepared } from './types';
+import { createTheme } from './createTheme';
+import { useTheme } from './useTheme';
+import * as classes from './ThemeProvider.scss';
+
+export type ThemeColorSet =
+  | Partial<{
+      default: string;
+      hover: string;
+      active: string;
+      disabled: string;
+    }>
+  | string;
+
+export type ThemePlateSet = Partial<{
+  fill: ThemeColorSet;
+  text: ThemeColorSet;
+  subText: ThemeColorSet;
+  link: ThemeColorSet;
+  divider: ThemeColorSet;
+}>;
+
+export type TokenSetType = string | { [key: string]: TokenSetType | undefined };
+
+export interface ThemeProviderProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The theme provided by the user.
+   */
+  theme?: Theme;
+}
+
+/**
+ * ThemeProvider, used for providing css variables and registering stylesheets.
+ */
+export const ThemeProvider = React.forwardRef<HTMLDivElement, ThemeProviderProps>(
+  (
+    { theme, style, className, ...rest }: React.PropsWithChildren<ThemeProviderProps>,
+    ref: React.Ref<HTMLDivElement>,
+  ) => {
+    // Pull contextual theme.
+    const parentTheme = useTheme();
+
+    // Merge the theme only when parent theme or props theme mutates.
+    const fullTheme = React.useMemo<ThemePrepared>(() => createTheme(parentTheme, theme), [parentTheme, theme]);
+
+    // Generate the inline style object only when merged theme mutates.
+    const inlineStyle = React.useMemo<React.CSSProperties>(
+      () => tokensToStyleObject(fullTheme.tokens, undefined, { ...style }),
+      [fullTheme, style],
```

Comment:
```suggestion
      [fullTheme.tokens, style],
```

Reply:
I don't think we need the extra bytes here. Comparing the object which I'm creating a few lines up should be enough, no?